### ISSUE TITLE
Problems Merging Fallout 4 plugins.

### ISSUE THREAD

**Issue by A-Sparowe** [matortheeternal/merge-plugins#133]
Let me first start off by saying that I am only experiencing this problem when trying to merge plugins for Fallout 4. Skyrim (Oldrim) works great and I have yet to mod SSE yet, so I have no idea if this problem carries over to another 64-bit game.

So when using Merge Plugins to bundle up my armor mods, most recent attempt was about 20 mods, or less. Everything works fine until I attempt to actually go through with the merge itself. I can check for errors and resolve errors without a problem but when I merge I will immediately get an error with "Robocopy" crashing. It crashes on every attempt to merge a mod from my list, meaning that the error popup box itself has to be clicked acknowledging the error for each one of the mods in my merge list. Once that is done spamming me with the error I can clearly see that the merge failed by checking the output folder which just contains a bunch of different TXT files for each failed merge attempt. 

Now I know this has nothing to do with the size of the merge I am attempting, I tried to merge two plugins together and I still got the same issue, because I've successfully merged A LOT more plugins for Skyrim (oldrim) without any issues or problems in game. All my weapons, armors, etc show up and to my knowledge it doesn't seem like there is anything bad going on behind the scenes since I tend to try and not merge any plugins that are for heavily scripted mods I am using. I know I can if I wanted to, but it's just something I do. The biggest merged I have right now for Skyrim (oldrim) is just over 80 Mods merged into a single esp file. All armor mods. 

I have no idea what "Robocopy" even is, I assume it's a cmd prompt to carry out stuff behind the scenes when batching my mods together? 

Also, I don't use NMM. Using the latest version of Mod Organizer 2 that supports Fallout 4 and now Skyrim SE. (So glad to see somebody picked up where Tanin left off)

I also use Mod Organizer (32bit) for Skyrim (oldrim) and have never had a problem running MP through MO to conduct all my merging needs. 

Hope this is enough information regarding my problem, but if you need more just let me know and I can attempt the merge again and zip up my output folder with whatever does get put in there, along with anything else you'd need to troubleshoot.

**Comment by matortheeternal** [comment]
robocopy is a system utility for copying files.  Merge Plugins compiles a batch script which uses the `robocopy` command to copy asset files to complete merges.

Are you using a separate version of Merge Plugins from the one that works with Skyrim?  You may need to give `MergePlugins.exe` administrative access from its properties window.  It's also possible that MO2 is interfering with Merge Plugins' ability to copy files using robocopy.  You can try disabling batch copying from the Merging tab of the Options Window as an alternative.

**Comment by A-Sparowe** [comment]
Sorry for the wall of text below but the first 2/3 of it explain in detail what I had to do to get around my problem and what I think was the overall cause of it. 

I'm using the version that's on the Skyrim Nexus page. I figured since it was updated for Fallout 4 and that there wasn't a separate version of the utility on the Fallout 4 Nexus that this was the one to use. I've already given it Admin Rights, as I did for MO2. 

I still haven't been able to successfully merge any plugins for FO4 using MP within MO2 ( newest version that supports FO4 and SSE ). I was able to "complete" the merges I made within MP by first clicking "OK" on all the pop-ups that appeared from the program regarding "Robocopy". Depending on the amount of mods in my merges, doing this was somewhat tedious. After that was done I left the output window in MP open and then navigated to the MP output directory for the merge I was working on and executed the "Batch" file found within the folder for the merge I was attempting. Doing this I was able to finish building my merges. I then closed the output window, followed by closing MP itself. Then I reloaded MP through MO2 to check if the merges were in fact "built" by heading over to the "Merges" tab within MP. They were in fact built and registered through MP as actual "Merged Plugins" 

Now this is only an educated guess based on what I have experienced and what I've come to know by using MP in the past for Skyrim (LE), but it seems as though MO2 is preventing MP from building merges when executing the program from MO2. So from what I can tell, it's not a problem on the MP side of things, though it might partially be? It seems to be more on the MO2 side of things and that maybe a patch from both development teams in the future will fix the issue. Since MO2 development has only recently started back up again, with new developers now that Tannin is working on NMM, I figure that in the future things will work themselves out. 

Now the last time I was actually playing FO4 and modding it was when the game first came out and I was doing this by using NMM. At this time I had no clue MP existed or that I could achieve something similar through using the script within xEdit for FO4. Actually FO4 is what got me into modding Skyrim and I completely stopped playing FO4 altogether, switched to MO and have been playing Skyrim since. 

I know that while Tannin was still working on MO and MO2 that there was support added for FO4 and I have no idea if the new team is building off of his past work or if they started from scratch. I can only assume that when MP added support for FO4 that it was when Tannin was still updating MO2. So maybe something got left out or overlooked by the new MO2 team regarding MP because I know not all features are working yet. 

Due to all these problems I was having with the new MO2 I gave up and downloaded NMM again to pick up FO4 again. This was only in the past week that I did this and I haven't tried using MP within NMM yet because I honestly haven't figured out how to get NMM to read it like it automatically does with xEdit and Bodyslide. I'll just have to watch GamerPoets video again explaining the installation process because I remember there being something in there about using MP with NMM. 

As far as MP and Skyrim SE goes, I have yet to attempt modding SSE since I'd rather wait til the 64-bit SKSE is made available, though I know it's possible to do so without SKSE. I honestly hate using NMM and really hope the new "Tannin NMM" version retains a lot of the older functionality of MO. 

BTW, love the new modpicker website you made. I'm hoping to add my lists and trials and tribulations to my profile soon. It's taking some time to get it all together but now that I see there's a new mod on the nexus for analyzing a users mods it might be easier for me lol! My left pane in MO has around 550-600 mods at the moment and this includes merges I made with MP. Some of these merges are rather big with my biggest being 80+ mods in one merge. These are all armor and or weapons put into merges. I haven't tried doing merges that deal with quests, textures, etc. I did merge all my player homes into a single merge, excluding a few of them that kept giving me errors when merging. Also merged all my followers into a single mod without problems so far. 

Sorry for the late response as well as my wall of text. I'm still rather new to the mod scene and at the same time I've been trying to learn how to use different programs for making my own content. 

Thanks again for all the work you've done for the community!
